src/kv/RocksDBStore.cc

```diff
@@ -646,10 +655,36 @@ rocksdb::ColumnFamilyHandle *RocksDBStore::get_cf_handle(const std::string& pref
     if (iter->second.handles.size() == 1) {
       return iter->second.handles[0];
     } else {
-      uint32_t hash_l = std::min<uint32_t>(iter->second.hash_l, keylen);
-      uint32_t hash_h = std::min<uint32_t>(iter->second.hash_h, keylen);
-      uint32_t hash = ceph_str_hash_rjenkins(&key[hash_l], hash_h - hash_l);
-      return iter->second.handles[hash % iter->second.handles.size()];
+      return get_key_cf(iter->second, key, keylen);
+    }
+  }
+}
+
+/**
+ * If the specified IteratorBounds arg has both an upper and a lower bound defined, and they have equal placement hash
+ * strings, we can be sure that the entire iteration range exists in a single CF. In that case, we return the relevant
+ * CF handle. In all other cases, we return a nullptr to indicate that the specified bounds cannot necessarily be mapped
+ * to a single CF.
```

Comment:
The comment here is a bit wrong.
It is of course true that for restricting search for one column family we must make sure that no key can squeeze between lower and upper bound and have different hash. 

To be sure that all keys in range [lower_bound .. upper_bound] belong to one shard, following must be true:
```
0000000000000000111111111
0123456789abcdef012345678
lower_bound_value
upper_bound_value
  ^--hash--^
```
1) lower_bound.size() >= hash_h
2) upper_bound.size() >= hash_h
3) lower_bound[0..hash_h] == upper_bound[0..hash_h]

Important part here is, that first hash_l characters cannot be ignored, as it would trigger as going to one shard:
```
0000000000000000111111111
0123456789abcdef012345678
ABC_bound_value_lower
DEF_bound_value_upper
    ^--hash--^
```

Reply:
I've updated the comment to include a note about that bit